lua/net.lua

```diff
@@ -1,17 +1,61 @@
 local gatekeeperc = require("gatekeeperc")
+local ffi = require("ffi")
+local ifaces = require("if_map")
 
 local M = {}
 
+function init_pci_addrs(iface, ports)
+	local pci_strs = ffi.new("const char *[" .. #ports .. "]")
+	for i, v in ipairs(ports) do
+		pci_strs[i - 1] = ifaces[v]
+	end
+	iface.pci_addrs = gatekeeperc.lua_init_pci_addrs(pci_strs, #ports)
+end
+
 -- Function that sets up the network.
 function M.setup_block()
 
 	-- Init the network configuration structure.
 	local conf = gatekeeperc.get_net_conf()
-	conf.num_rx_queues = 1
-	conf.num_tx_queues = 1
+
+	-- Change these parameters to configure the network.
+	local front_ports = {"enp131s0f0"}
+	local back_ports = {"enp133s0f0"}
+	conf.num_rx_queues = 2
+	conf.num_tx_queues = 2
+
+	-- Code below this point should not need to be changed.
+	conf.front.num_ports = #front_ports
+	conf.front.name = gatekeeperc.lua_init_iface_name("front")
```

Comment:
After reviewing the code, I think all these code trying to do is to initialize the interface. Instead of exposing a set of C functions to finish the initialization, I wonder whether we can just expose one C function, e.g., ``init_iface("front", #front_ports, port_names)``, where ``port_names`` is the ``const char *`` array containing ``enp131s0f0``. Thus, we may not need the separate file `` generate_if_map.c``,  instead we can maintain the mapping function in lib/net.c. Therefore, this C function actually initializes the front interface.